presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -2812,6 +2812,59 @@ protected void testUpdatePartitionStatistics(
                 .isEqualTo(ImmutableMap.of(firstPartitionName, initialStatistics, secondPartitionName, initialStatistics));
     }
 
+    /**
+     * This test creates 2 identical partitions and verifies that the statistics projected based on
+     * a single partition sample are equal to the statistics computed in a fair way
+     */
+    @Test
+    public void testPartitionStatisticsSampling()
+            throws Exception
+    {
+        SchemaTableName tableName = temporaryTable("test_partition_statistics_sampling");
+
+        try {
+            createDummyPartitionedTable(tableName, STATISTICS_PARTITIONED_TABLE_COLUMNS);
+
+            metastoreClient.updatePartitionStatistics(tableName.getSchemaName(), tableName.getTableName(), "ds=2016-01-01", actualStatistics -> STATISTICS_1);
+            metastoreClient.updatePartitionStatistics(tableName.getSchemaName(), tableName.getTableName(), "ds=2016-01-02", actualStatistics -> STATISTICS_1);
+
+            try (Transaction transaction = newTransaction()) {
+                ConnectorSession session = newSession();
+                ConnectorMetadata metadata = transaction.getMetadata();
+
+                ConnectorTableHandle tableHandle = metadata.getTableHandle(session, tableName);
+                Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
+
+                TableStatistics expectedStatistics = metadata.getTableStatistics(sampleSize(2), tableHandle, Constraint.alwaysTrue());
```

Comment:
it is still `actual` (not `expected`) because value is calculated by program not hardcoded by you, so please rename it. Maybe just `statistics`?

Reply:
or `unsampledStatistics`